lib/ExpensiMark.js

```diff
@@ -330,6 +330,24 @@ export default class ExpensiMark {
                 regex: /<\/h1><br\s*[/]?>/gi,
                 replacement: '</h1>',
             },
+            {
+                // This rule is designed to address cases where HTML content ends up within attribute values,
+                // HTML can end up within attributes, because different rules are applied in sequence
+                // To ensure cleanliness and correctness of HTML attributes, this cleanup rule is executed last.
+                // We're limiting the scope of this rule to 'img' tags and their 'alt' attributes, which are the
+                // primary sources of such issues. This approach keeps the operation efficient and focused.
+                // However, the pattern is extensible to additional tags and attributes as needed.
+                name: 'attributeCleanup',
+                regex: /<(?:img)[^>]*\s(?:alt)=(["'])([^"']*)\1[^>]*>/gi,
+                replacement: (match, g1, g2) => {
+                    const cleanedG2 = this.htmlToMarkdown(g2);
+                    if (cleanedG2 === g2) {
+                        return match;
+                    }
```

Comment:
Do we need this? What happens if we always return `match.replace(g2, _.escape(cleanedG2))` instead? 
Will it be a performance issue or something else?
Because if its a performance issue wont it be faster to check if there are any markdowns to convert if none return g2 early. But if its not a performance concern won't it be more readable skipping this and always returning `match.replace(g2, _.escape(cleanedG2))`.

Reply:
Since `g2` can include HTML content, and I'm not aware of a quick function to check for HTML markup, utilizing the parser allows us to both detect HTML and process the content for the next step efficiently.
The code, while a bit verbose, optimizes performance slightly - the `_.escape()` function is called only when necessary.

If you wish to shorten it to one line, just say so and I'll update it when I get the chance